src/gsi/setupq.f90

```diff
@@ -244,6 +248,7 @@ subroutine setupq(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,conv_diagsav
   character(8) station_id
   character(8),allocatable,dimension(:):: cdiagbuf,cdiagbufp
   character(8),allocatable,dimension(:):: cprvstg,csprvstg
+  character(8),allocatable,dimension(:):: cprvstgp,csprvstgp ! <-- provider info array for pseudo obs
```

Comment:
Please rethink about if the provider information from pseudo observations are need in this situation.
The pseudo observation has the same innovation as the surface observation it derived from. It will not provide more information for calculating the the uselist. I suggest to make cprvstgp,csprvstgp a location variable and fill with missing values if it is pseudo observation.